### ISSUE TITLE
&Itemid= is not being added to url for "deep" content

### ISSUE THREAD

**Issue by williamconifer** [getk2/k2-links-for-jce#5]
Greetings. 

When using K2 Links for JCE we are not getting &Itemid=number added to the url for k2 items that do not have a menu item associated to them in Joomla when we use the folder tree navigation in your plugin. 

We use menus to "deliver" a visitor to a K2 category. The visitor then uses the K2 category system to dig down to the K2 item they want. So if Joomla menu item id 256 delivers the visitor to Category ABC then I would want all K2 items under that category or subcategory to have &Itemid=256 in the url when I use your plugin. THIS is not happening when we use the folder tree navigation in your plugin.  

HOWEVER, if I use the Search tool (which is great!) in K2 Links for JCE and find the K2 Item I want, it INCLUDES the &Itemid=number in the URL. 

To summarize: using Search to find the K2 item will include the &Itemid=number BUT if I navigate to the K2 item using your folder tree it does NOT include the &Itemid=number.

Any chance of getting this fixed? I have a couple of city government websites that is really content heavy with lots of users. 

Thanks
jack

**Comment by kavadas** [comment]
Hi,

It seems that you have confused some things here. The search feature was added in the latest versions of JCE and is not supported at all by this plugin. The results you get there are links to Joomla! content articles, not K2 items.

Regarding the actual issue, K2 will assign an Itemid to the links as long as you have a menu link pointing to that item or to it's category. I suggest to create a menu link to all categories so any non-mapped items will map under that menu link.

This is the way K2 detects the links, and has nothing to do with that plugin. That plugin is just using K2 internals to detect the links.
